sample_consensus/include/pcl/sample_consensus/impl/sac_model_circle.hpp

```diff
@@ -156,13 +162,10 @@ pcl::SampleConsensusModelCircle2D<PointT>::selectWithinDistance (
     if (distance < threshold)
     {
       // Returns the indices of the points whose distances are smaller than the threshold
-      inliers[nr_p] = (*indices_)[i];
-      error_sqr_dists_[nr_p] = static_cast<double> (distance);
-      ++nr_p;
+      inliers.push_back ((*indices_)[i]);
```

Comment:
Some loops can be modified to remove the loop variable `i` and replaced by a range-based for loop. Not all though. Might be something to add

Reply:
Yes, that would be possible. But I think for this PR I will leave it like it is.